docs/articles/vector_search_based_recommender_system.md

```diff
@@ -0,0 +1,276 @@
+            model="text-embedding-ada-002"
+            )
+        search_query_embedding = response.data[0].embedding
+        openai_end = (datetime.now())
+        elapsed_time = openai_end - openai_start
+
+        return search_query_embedding
+
+    ```
+    3. get_rs_results(): Utilizes Query Lambda created earlier in Rockset and returns recommendations based on user inputs and embeddings.
+    ```python
+
+    def get_rs_results(inputs, region, rockset_key, search_query_embedding):
+        print("\nRunning Rockset Queries...")
+        # Create an instance of the Rockset client
+        rs = RocksetClient(api_key=rockset_key, host=region)
+
+        rockset_start = (datetime.now())
+        # Execute Query Lambda By Version
+        rockset_start = (datetime.now())
+        api_response = rs.QueryLambdas.execute_query_lambda_by_tag(
+            workspace="commons",
+            query_lambda="recommend_games",
+            tag="latest",
+            parameters=[
+                {
+                    "name": "embedding",
+                    "type": "array",
+                    "value": str(search_query_embedding)
+                },
+                {
+                    "name": "min_price",
+                    "type": "int",
+                    "value": inputs["min_price"]
+                },
+                {
+                    "name": "max_price",
+                    "type": "int",
+                    "value": inputs["max_price"]
+                },
+                {
+                    "name": "brand",
+                    "type": "string",
+                    "value": inputs["brand"]
+                }
+                {
+                     "name": "limit",
+                     "type": "int",
+                     "value": inputs["limit"]
+                }
+            ]
+        )
+        rockset_end = (datetime.now())
+        elapsed_time = rockset_end - rockset_start
+
+        records_list = []
+        for record in api_response["results"]:
+            record_data = {
+                "title": record['title'],
+                "image_url": record['image_ur1'],
+                "brand": record['brand'],
+                "estimated_price": record['estimated_price'],
+                "description": record['description']
+            }
+            records_list.append(record_data)
+
+        return records_list
+    ```
+
+Overall, the Flask backend processes user input and interacts with external services (OpenAI and Rockset) to provide dynamic content to the frontend. It extracts form data from the frontend, generates OpenAI embeddings for text queries, and utilizes Query Lambda at Rockset to find recommendations.
+
+
+Now, you are ready to run the flask server and access it via your internet browser. Our application is up and running. Let's add some parameters in the bar and get some recommendations. The results will be displayed on an HTML template as shown below.
+
```

Comment:
"in the bar" = ?

Reply:
search bar. If you look into the frontend picture attached, you'll understand that the reference was to the search bar on the left.